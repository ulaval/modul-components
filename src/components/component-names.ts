export const ACCORDION_GROUP_NAME: string = 'm-accordion-group';
export const ACCORDION_NAME: string = 'm-accordion';
export const LIST_ITEM_NAME: string = 'm-list-item';
export const BUTTON_NAME: string = 'm-button';
export const BUTTON_GROUP_NAME: string = 'm-button-group';
export const CAROUSEL_NAME: string = 'm-carousel';
export const CAROUSEL_ITEM_NAME: string = 'm-carousel-item';
export const CHECKBOX_NAME: string = 'm-checkbox';
export const DATEFIELDS_NAME: string = 'm-datefields';
export const DATEPICKER_NAME: string = 'm-datepicker';
export const DIALOG_NAME: string = 'm-dialog';
export const DROPDOWN_NAME: string = 'm-dropdown';
export const DROPDOWN_ITEM_NAME: string = 'm-dropdown-item';
export const DROPDOWN_GROUP_NAME: string = 'm-dropdown-group';
export const DYNAMIC_TEMPLATE_NAME: string = 'm-dynamic-template';
export const FILE_SELECT_NAME: string = 'm-file-select';
export const FILE_UPLOAD_NAME: string = 'm-file-upload';
export const EDIT_WINDOW_NAME: string = 'm-edit-window';
export const FLEX_TEMPLATE_NAME: string = 'm-flex-template';
export const I18N_NAME: string = 'm-i18n';
export const ICON_NAME: string = 'm-icon';
export const ICON_BUTTON_NAME: string = 'm-icon-button';
export const INPUT_STYLE_NAME: string = 'm-input-style';
export const LIMIT_TEXT_NAME: string = 'm-limit-text';
export const LINK_NAME: string = 'm-link';
export const MESSAGE_NAME: string = 'm-message';
export const MODAL_NAME: string = 'm-modal';
export const NAVBAR_NAME: string = 'm-navbar';
export const NAVBAR_ITEM_NAME: string = 'm-navbar-item';
export const MENU_NAME: string = 'm-menu';
export const MENU_ITEM_NAME: string = 'm-menu-item';
export const PANEL_NAME: string = 'm-panel';
export const PHONE_NUMBER_NAME: string = 'm-phone-number';
export const POPPER_NAME: string = 'm-popper';
export const POPUP_NAME: string = 'm-popup';
export const PROGRESS_NAME: string = 'm-progress';
export const RADIO_NAME: string = 'm-radio';
export const RADIO_GROUP_NAME: string = 'm-radio-group';
export const RADIO_STYLE_NAME: string = 'm-radio-style';
export const SCROLL_TOP_NAME: string = 'm-scroll-top';
export const SIDEBAR_NAME: string = 'm-sidebar';
export const SLIDER_NAME: string = 'm-slider';
export const SPINNER_NAME: string = 'm-spinner';
export const STATUS_NAME: string = 'm-status';
export const STEP_NAME: string = 'm-step';
export const STEPPERS_NAME: string = 'm-steppers';
export const STEPPERS_ITEM_NAME: string = 'm-steppers-item';
export const SWITCH_NAME: string = 'm-switch';
export const TAB_PANEL_NAME: string = 'm-tab-panel';
export const TABS_NAME: string = 'm-tabs';
export const TEMPLATE_NAME: string = 'm-template';
export const TEXTAREA_NAME: string = 'm-textarea';
export const TEXTFIELD_NAME: string = 'm-textfield';
export const TIMEPICKER_NAME: string = 'm-timepicker';
export const TOOLTIP_NAME: string = 'm-tooltip';
export const UPLOAD_NAME: string = 'm-upload';
export const UPLOAD_INPUT_NAME: string = 'm-upload-input';
export const UPLOAD_DRAGDROP_NAME: string = 'm-upload-dragdrop';
export const UPLOAD_FILESLIST_NAME: string = 'm-upload-fileslist';
export const VALIDATION_MESSAGE_NAME: string = 'm-validation-message';
