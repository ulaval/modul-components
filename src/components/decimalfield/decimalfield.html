<div class="m-decimalfield"
     :class="[{ 'm--is-disabled': isDisabled,
                'm--is-waiting': isWaiting,
                'm--is-readonly': isReadonly,
                'm--has-error': hasDecimalfieldError,
                'm--is-valid': isDecimalfieldValid,
                'm--has-label': hasLabel,
                'm--has-validation-message': hasValidationMessage } ]"
     @click="onClick"
     :style="{ width: inputWidth, maxWidth: inputMaxWidth }">
    <m-input-style :disabled="isDisabled"
                   :waiting="isWaiting"
                   :error="hasDecimalfieldError"
                   :valid="isDecimalfieldValid"
                   :label="label"
                   :label-for="id"
                   :focus="isFocus"
                   :label-up="labelUp"
                   :empty="isEmpty"
                   :readonly="isReadonly"
                   :required-marker="requiredMarker"
                   :tag-style="tagStyle">
        <template slot="prefix">
            <slot name="prefix"></slot>
        </template>
        <m-input-mask class="m-decimalfield__input"
                      v-model="model"
                      :id="id"
                      :disabled="isDisabled || isWaiting"
                      :placeholder="placeholder"
                      :readonly="isReadonly"
                      @focus="onFocus"
                      @blur="onBlur"
                      @keyup="onKeyup"
                      @change="onChange"
                      ref="inputMask"
                      :options="inputMaskOptions"
                      inputmode="decimal"></m-input-mask>
        <template slot="suffix">
            <slot name="suffix"></slot>
        </template>
    </m-input-style>
    <div class="m-decimalfield__validation">
        <m-validation-message class="m-decimalfield__validation__message"
                              :disabled="isDisabled"
                              :waiting="isWaiting"
                              :error="hasDecimalfieldError"
                              :error-message="errorMessage"
                              :valid="isDecimalfieldValid"
                              :valid-message="validMessage"
                              :helper-message="helperMessage"></m-validation-message>
    </div>
</div>
