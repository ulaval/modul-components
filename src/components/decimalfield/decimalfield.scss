@import 'commons';

.m-decimalfield {
    position: relative;
    display: inline-flex;
    flex-direction: column;

    @include m-input-inline-spacing();

    &.m--has-validation-message {
        .m-decimalfield__validation {
            margin-top: $m-spacing--xs;
        }
    }

    &.m--has-error {
        .m-decimalfield__validation {
            color: $m-color--error;
        }
    }

    &.m--is-valid {
        .m-decimalfield__validation {
            color: $m-color--success;
        }
    }

    &__validation {
        transition: margin-top $m-transition-duration ease;
        display: flex;
        justify-content: flex-end;

        &__message {
            flex: 1 1 auto;
        }
    }

    &__input {
        appearance: textfield;

        &::-webkit-outer-spin-button,
        &::-webkit-inner-spin-button {
            appearance: textfield;
        }
    }
}
