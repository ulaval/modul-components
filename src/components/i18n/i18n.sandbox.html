<div>
    <h5>default</h5>
    <div style="border: solid 1px black">
        <m-i18n></m-i18n>
    </div>

    <h5>sample</h5>
    <m-i18n></m-i18n>
</div>
