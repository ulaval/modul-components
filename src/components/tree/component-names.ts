export const TREE_NODE_NAME: string = 'm-tree-node';
export const TREE_ICON_NAME: string = 'm-tree-icon';
