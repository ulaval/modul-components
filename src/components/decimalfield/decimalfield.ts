import { PluginObject } from 'vue';
import Component from 'vue-class-component';
import { Emit, Prop } from 'vue-property-decorator';
import { InputLabel } from '../../mixins/input-label/input-label';
import { InputManagement } from '../../mixins/input-management/input-management';
import { InputState } from '../../mixins/input-state/input-state';
import { InputWidth } from '../../mixins/input-width/input-width';
import L10nPlugin, { MDecimalFormat } from '../../utils/l10n/l10n';
import uuid from '../../utils/uuid/uuid';
import { ModulVue } from '../../utils/vue/vue';
import { DECIMALFIELD_NAME } from '../component-names';
import { InputMaskOptions, MInputMask } from '../input-mask/input-mask';
import InputStyle from '../input-style/input-style';
import ValidationMesagePlugin from '../validation-message/validation-message';
import WithRender from './decimalfield.html?style=./decimalfield.scss';

@WithRender
@Component({
    mixins: [
        InputState,
        InputWidth,
        InputLabel,
        InputManagement
    ],
    components: {
        MInputMask
    }
})
export class MDecimalfield extends ModulVue {
    @Prop()
    public value: number;

    @Prop({
        default: 2
    })
    public rounding: number;

    @Prop({
        default: 14
    })
    public precision: number;

    @Prop({ default: false })
    public forceRoundingFormat: boolean;

    protected id: string = `mDecimalfield-${uuid.generate()}`;

    private get hasDecimalfieldError(): boolean {
        return this.as<InputState>().hasError;
    }

    private get isDecimalfieldValid(): boolean {
        return this.as<InputState>().isValid;
    }

    private get inputMaskOptions(): InputMaskOptions {
        const decimalFormat: MDecimalFormat = this.$l10n.getDecimalFormat(this.currentLocale);
        return {
            numeral: true,
            numeralThousandsGroupStyle: 'thousand',
            numeralIntegerScale: this.precision - this.rounding,
            numeralDecimalScale: this.rounding,
            numeralDecimalMark: decimalFormat.decimalMark,
            numeralPositiveOnly: true,
            stripLeadingZeroes: true,
            delimiter: decimalFormat.thousandSeparator,
            removeTrailingDecimalMark: true,
            forceDecimalScale: this.forceRoundingFormat
        };
    }

    get currentLocale(): string {
        return (this as ModulVue).$i18n.currentLocale;
    }

    private get model(): string {
        return (!this.value && this.value !== 0 ? '' : this.value).toString();
    }

    private set model(value: string) {
        const valueAsNumber: number = Number.parseFloat(value);
        this.emitNewValue(valueAsNumber);
    }

    @Emit('input')
    emitNewValue(_newValue: number): void { }
}

const DecimalfieldPlugin: PluginObject<any> = {
    install(v): void {
        v.use(L10nPlugin);
        v.use(InputStyle);
        v.use(ValidationMesagePlugin);
        v.component(DECIMALFIELD_NAME, MDecimalfield);
    }
};

export default DecimalfieldPlugin;
