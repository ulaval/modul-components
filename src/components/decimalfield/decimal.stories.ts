import { withA11y } from '@storybook/addon-a11y';
import { select, withKnobs } from '@storybook/addon-knobs';
import { storiesOf } from '@storybook/vue';
import Vue from 'vue';
import { componentsHierarchyRootSeparator } from '../../../conf/storybook/utils';
import { ENGLISH, FRENCH } from '../../utils/i18n/i18n';
import { ModulVue } from '../../utils/vue/vue';
import { DECIMALFIELD_NAME } from '../component-names';
import { MDecimalfield } from './decimalfield';

storiesOf(`${componentsHierarchyRootSeparator}${DECIMALFIELD_NAME}`, module)
    .addDecorator(withA11y)
    .addDecorator(withKnobs)
    .add('Basic', () => ({
        components: { MDecimalfield },
        template: `
            <div>
                <div><${DECIMALFIELD_NAME} v-model="value"></${DECIMALFIELD_NAME}></div>
                <div><${DECIMALFIELD_NAME} v-model="value" :label="'Label'"></${DECIMALFIELD_NAME}></div>
                <div><${DECIMALFIELD_NAME} v-model="value" :label="'Required label'" :required-marker="true"></${DECIMALFIELD_NAME}></div>
                <div><${DECIMALFIELD_NAME} v-model="value" :valid-message="'Valid message'"></${DECIMALFIELD_NAME}></div>
                <div><${DECIMALFIELD_NAME} v-model="value" :helper-message="'Helper message'"></${DECIMALFIELD_NAME}></div>
                <div><${DECIMALFIELD_NAME} v-model="value" :error-message="'Error message'"></${DECIMALFIELD_NAME}></div>
            </div>
        `,
        data: () => ({
            value: undefined
        })
    }));

storiesOf(`${componentsHierarchyRootSeparator}${DECIMALFIELD_NAME}`, module)
    .addDecorator(withA11y)
    .addDecorator(withKnobs)
    .add('States', () => ({
        components: { MDecimalfield },
        template: `
            <div>
                <div><${DECIMALFIELD_NAME} v-model="value" :disabled="true" :placeholder="'Disabled'"></${DECIMALFIELD_NAME}></div>
                <div><${DECIMALFIELD_NAME} v-model="value" :placeholder="'Read-only'" :readonly="true"></${DECIMALFIELD_NAME}></div>
                <div><${DECIMALFIELD_NAME} v-model="value" :placeholder="'Waiting'" :waiting="true"></${DECIMALFIELD_NAME}></div>
            </div>
        `,
        data: () => ({
            value: undefined
        })
    }));

storiesOf(`${componentsHierarchyRootSeparator}${DECIMALFIELD_NAME}`, module)
    .addDecorator(withA11y)
    .addDecorator(withKnobs)
    .add('With initial value (0)', () => ({
        components: { MDecimalfield },
        template: `<${DECIMALFIELD_NAME} v-model="value"></${DECIMALFIELD_NAME}>`,
        data: () => ({
            value: 0
        })
    }));

storiesOf(`${componentsHierarchyRootSeparator}${DECIMALFIELD_NAME}`, module)
    .addDecorator(withA11y)
    .addDecorator(withKnobs)
    .add('With initial value', () => ({
        components: { MDecimalfield },
        template: `<${DECIMALFIELD_NAME} v-model="value"></${DECIMALFIELD_NAME}>`,
        data: () => ({
            value: 123456.78
        })
    }));

storiesOf(`${componentsHierarchyRootSeparator}${DECIMALFIELD_NAME}`, module)
    .addDecorator(withA11y)
    .addDecorator(withKnobs)
    .add('Localization', () => ({
        components: { MDecimalfield },
        template: `
            <${DECIMALFIELD_NAME} v-model="value"></${DECIMALFIELD_NAME}>
        `,
        props: {
            lang: {
                default: select('Language', {
                    [FRENCH]: FRENCH,
                    [ENGLISH]: ENGLISH
                }, (Vue.prototype as ModulVue).$i18n.currentLang())
            }
        },
        watch: {
            lang: function(): void {
                this.i18n.currentLang(this.lang);
            }
        },
        data: () => ({
            value: 123456.78,
            originalLang: (Vue.prototype as ModulVue).$i18n.currentLang(),
            i18n: Vue.prototype.$i18n
        }),
        destroyed(): void {
            (Vue.prototype as ModulVue).$i18n.currentLang(this.originalLang);
        }
    }));

storiesOf(`${componentsHierarchyRootSeparator}${DECIMALFIELD_NAME}`, module)
    .addDecorator(withA11y)
    .addDecorator(withKnobs)
    .add('Default precision', () => ({
        components: { MDecimalfield },
        template: `
            <${DECIMALFIELD_NAME} v-model="value"></${DECIMALFIELD_NAME}>
        `,
        data: () => ({
            value: 9999999999999.99
        })
    }))
    .add('Custom precision', () => ({
        components: { MDecimalfield },
        template: `
            <${DECIMALFIELD_NAME} v-model="value" :precision="9" :rounding="4"></${DECIMALFIELD_NAME}>
        `,
        data: () => ({
            value: 99999.9999
        })
    }));
