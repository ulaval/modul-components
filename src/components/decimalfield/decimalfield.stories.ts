import { select } from '@storybook/addon-knobs';
import { storiesOf } from '@storybook/vue';
import Vue from 'vue';
import { componentsHierarchyRootSeparator } from '../../../conf/storybook/utils';
import { ENGLISH, FRENCH } from '../../utils/i18n/i18n';
import { ModulVue } from '../../utils/vue/vue';
import { DECIMALFIELD_NAME } from '../component-names';
import DecimalfieldPlugin, { MDecimalfield } from './decimalfield';

Vue.use(DecimalfieldPlugin);


storiesOf(`${componentsHierarchyRootSeparator}${DECIMALFIELD_NAME}`, module)


    .add('Basic', () => ({
        components: { MDecimalfield },
        template: `
            <div>
                <div><${DECIMALFIELD_NAME} v-model="value"></${DECIMALFIELD_NAME}></div>
                <div><${DECIMALFIELD_NAME} v-model="value" :label="'Label'"></${DECIMALFIELD_NAME}></div>
                <div><${DECIMALFIELD_NAME} v-model="value" :label="'Required label'" :required-marker="true"></${DECIMALFIELD_NAME}></div>
                <div><${DECIMALFIELD_NAME} v-model="value" :valid-message="'Valid message'"></${DECIMALFIELD_NAME}></div>
                <div><${DECIMALFIELD_NAME} v-model="value" :helper-message="'Helper message'"></${DECIMALFIELD_NAME}></div>
                <div><${DECIMALFIELD_NAME} v-model="value" :error-message="'Error message'"></${DECIMALFIELD_NAME}></div>
                <div><${DECIMALFIELD_NAME} v-model="value" :label="'Label'" :placeholder="'Enter a decimal number'" :label-up="true"></${DECIMALFIELD_NAME}></div>
            </div>
        `,
        data: () => ({
            value: undefined
        })
    }));

storiesOf(`${componentsHierarchyRootSeparator}${DECIMALFIELD_NAME}`, module)


    .add('States', () => ({
        components: { MDecimalfield },
        template: `
            <div>
                <div><${DECIMALFIELD_NAME} v-model="value" :disabled="true" :placeholder="'Disabled'"></${DECIMALFIELD_NAME}></div>
                <div><${DECIMALFIELD_NAME} v-model="value" :placeholder="'Read-only'" :readonly="true"></${DECIMALFIELD_NAME}></div>
                <div><${DECIMALFIELD_NAME} v-model="value" :placeholder="'Waiting'" :waiting="true"></${DECIMALFIELD_NAME}></div>
            </div>
        `,
        data: () => ({
            value: undefined
        })
    }));

storiesOf(`${componentsHierarchyRootSeparator}${DECIMALFIELD_NAME}`, module)


    .add('With initial value (0)', () => ({
        components: { MDecimalfield },
        template: `<${DECIMALFIELD_NAME} v-model="value"></${DECIMALFIELD_NAME}>`,
        data: () => ({
            value: 0
        })
    }));

storiesOf(`${componentsHierarchyRootSeparator}${DECIMALFIELD_NAME}`, module)


    .add('With initial value', () => ({
        components: { MDecimalfield },
        template: `<${DECIMALFIELD_NAME} v-model="value"></${DECIMALFIELD_NAME}>`,
        data: () => ({
            value: 123456.1
        })
    }));

storiesOf(`${componentsHierarchyRootSeparator}${DECIMALFIELD_NAME}`, module)


    .add('Localization', () => ({
        components: { MDecimalfield },
        template: `
            <${DECIMALFIELD_NAME} v-model="value"></${DECIMALFIELD_NAME}>
        `,
        props: {
            lang: {
                default: select('Language', {
                    [FRENCH]: FRENCH,
                    [ENGLISH]: ENGLISH
                }, (Vue.prototype as ModulVue).$i18n.currentLang())
            }
        },
        watch: {
            lang: function(): void {
                this.i18n.currentLang(this.lang);
            }
        },
        data: () => ({
            value: 123456.1,
            originalLang: (Vue.prototype as ModulVue).$i18n.currentLang(),
            i18n: Vue.prototype.$i18n
        }),
        destroyed(): void {
            (Vue.prototype as ModulVue).$i18n.currentLang(this.originalLang);
        }
    }));

storiesOf(`${componentsHierarchyRootSeparator}${DECIMALFIELD_NAME}`, module)


    .add('Default precision', () => ({
        components: { MDecimalfield },
        template: `
            <${DECIMALFIELD_NAME} v-model="value"></${DECIMALFIELD_NAME}>
        `,
        data: () => ({
            value: 9999999999999.99
        })
    }))
    .add('Custom precision', () => ({
        components: { MDecimalfield },
        template: `
            <${DECIMALFIELD_NAME} v-model="value" :precision="9" :rounding="4"></${DECIMALFIELD_NAME}>
        `,
        data: () => ({
            value: 99999.9999
        })
    }));
