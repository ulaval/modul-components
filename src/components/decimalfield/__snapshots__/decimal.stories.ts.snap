// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots components|m-decimalfield Basic 1`] = `
<div>
  <div>
    <div
      class="m-decimalfield"
      style="width: 100%; max-width: 288px;"
    >
      <div
        class="m-input-style m--is-tag-default"
      >
        <div
          class="m-input-style__content"
          style="margin-top: 8px;"
        >
          <!---->
           
          <div
            class="m-input-style__body"
          >
            <div
              class="m-input-style__prefix"
            />
             
            <div
              class="m-input-style__left-content"
            >
               
              <input
                class="m-decimalfield__input"
                id="mDecimalfield-uuid"
                inputmode="numeric"
              />
                
              <!---->
               
              <!---->
            </div>
             
            <div
              class="m-input-style__right-content"
            />
             
            <!---->
          </div>
        </div>
      </div>
       
      <div
        class="m-decimalfield__validation"
      >
        <div
          class="m-validation-message m-decimalfield__validation__message"
        >
          <!---->
           
          <!---->
        </div>
      </div>
    </div>
  </div>
   
  <div>
    <div
      class="m-decimalfield m--has-label"
      style="width: 100%; max-width: 288px;"
    >
      <div
        class="m-input-style m--has-label m--is-tag-default"
      >
        <div
          class="m-input-style__content"
          style="margin-top: 8px;"
        >
          <label
            class="m-input-style__label"
            for="mDecimalfield-uuid"
          >
            <span
              class="m-input-style__text"
            >
              <span>
                Label
              </span>
               
              <!---->
            </span>
          </label>
           
          <div
            class="m-input-style__body"
          >
            <!---->
             
            <div
              class="m-input-style__left-content"
            >
               
              <input
                class="m-decimalfield__input"
                id="mDecimalfield-uuid"
                inputmode="numeric"
              />
                
              <!---->
               
              <!---->
            </div>
             
            <div
              class="m-input-style__right-content"
            />
             
            <!---->
          </div>
        </div>
      </div>
       
      <div
        class="m-decimalfield__validation"
      >
        <div
          class="m-validation-message m-decimalfield__validation__message"
        >
          <!---->
           
          <!---->
        </div>
      </div>
    </div>
  </div>
   
  <div>
    <div
      class="m-decimalfield m--has-label"
      style="width: 100%; max-width: 288px;"
    >
      <div
        class="m-input-style m--has-label m--is-tag-default"
      >
        <div
          class="m-input-style__content"
          style="margin-top: 8px;"
        >
          <label
            class="m-input-style__label"
            for="mDecimalfield-uuid"
          >
            <span
              class="m-input-style__text"
            >
              <span>
                Required label
              </span>
               
              <sup
                class="m-input-style__required-marker"
              >
                *
              </sup>
            </span>
          </label>
           
          <div
            class="m-input-style__body"
          >
            <!---->
             
            <div
              class="m-input-style__left-content"
            >
               
              <input
                class="m-decimalfield__input"
                id="mDecimalfield-uuid"
                inputmode="numeric"
              />
                
              <!---->
               
              <!---->
            </div>
             
            <div
              class="m-input-style__right-content"
            />
             
            <!---->
          </div>
        </div>
      </div>
       
      <div
        class="m-decimalfield__validation"
      >
        <div
          class="m-validation-message m-decimalfield__validation__message"
        >
          <!---->
           
          <!---->
        </div>
      </div>
    </div>
  </div>
   
  <div>
    <div
      class="m-decimalfield m--is-valid m--has-validation-message"
      style="width: 100%; max-width: 288px;"
    >
      <div
        class="m-input-style m--is-valid m--is-tag-default"
      >
        <div
          class="m-input-style__content"
          style="margin-top: 8px;"
        >
          <!---->
           
          <div
            class="m-input-style__body"
          >
            <div
              class="m-input-style__prefix"
            />
             
            <div
              class="m-input-style__left-content"
            >
               
              <input
                class="m-decimalfield__input"
                id="mDecimalfield-uuid"
                inputmode="numeric"
              />
                
              <!---->
               
              <!---->
            </div>
             
            <div
              class="m-input-style__right-content"
            />
             
            <!---->
          </div>
        </div>
      </div>
       
      <div
        class="m-decimalfield__validation"
      >
        <div
          class="m-validation-message m-decimalfield__validation__message"
        >
          <p
            class="m-validation-message__valid"
          >
            <svg
              class="m-icon m-validation-message__icon"
              height="1em"
              width="1em"
            >
              <title>
                Valide
              </title>
               
              <use
                aria-hidden="true"
                class=""
              />
            </svg>
             
            <span
              class="m-validation-message__text"
            >
              Valid message
            </span>
          </p>
           
          <!---->
        </div>
      </div>
    </div>
  </div>
   
  <div>
    <div
      class="m-decimalfield m--has-validation-message"
      style="width: 100%; max-width: 288px;"
    >
      <div
        class="m-input-style m--is-tag-default"
      >
        <div
          class="m-input-style__content"
          style="margin-top: 8px;"
        >
          <!---->
           
          <div
            class="m-input-style__body"
          >
            <div
              class="m-input-style__prefix"
            />
             
            <div
              class="m-input-style__left-content"
            >
               
              <input
                class="m-decimalfield__input"
                id="mDecimalfield-uuid"
                inputmode="numeric"
              />
                
              <!---->
               
              <!---->
            </div>
             
            <div
              class="m-input-style__right-content"
            />
             
            <!---->
          </div>
        </div>
      </div>
       
      <div
        class="m-decimalfield__validation"
      >
        <div
          class="m-validation-message m-decimalfield__validation__message"
        >
          <!---->
           
          <p
            class="m-validation-message__helper"
          >
            Helper message
          </p>
        </div>
      </div>
    </div>
  </div>
   
  <div>
    <div
      class="m-decimalfield m--has-error m--has-validation-message"
      style="width: 100%; max-width: 288px;"
    >
      <div
        class="m-input-style m--has-error m--is-tag-default"
      >
        <div
          class="m-input-style__content"
          style="margin-top: 8px;"
        >
          <!---->
           
          <div
            class="m-input-style__body"
          >
            <div
              class="m-input-style__prefix"
            />
             
            <div
              class="m-input-style__left-content"
            >
               
              <input
                class="m-decimalfield__input"
                id="mDecimalfield-uuid"
                inputmode="numeric"
              />
                
              <!---->
               
              <!---->
            </div>
             
            <div
              class="m-input-style__right-content"
            />
             
            <!---->
          </div>
        </div>
      </div>
       
      <div
        class="m-decimalfield__validation"
      >
        <div
          class="m-validation-message m-decimalfield__validation__message"
        >
          <p
            class="m-validation-message__error"
          >
            <svg
              class="m-icon m-validation-message__icon"
              height="1em"
              width="1em"
            >
              <title>
                Erreur
              </title>
               
              <use
                aria-hidden="true"
                class=""
              />
            </svg>
             
            <span
              class="m-validation-message__text"
            >
              Error message
            </span>
          </p>
           
          <!---->
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Storyshots components|m-decimalfield Custom precision 1`] = `
<div
  class="m-decimalfield"
  style="width: 100%; max-width: 288px;"
>
  <div
    class="m-input-style m--has-value m--is-tag-default"
  >
    <div
      class="m-input-style__content"
      style="margin-top: 8px;"
    >
      <!---->
       
      <div
        class="m-input-style__body"
      >
        <div
          class="m-input-style__prefix"
        />
         
        <div
          class="m-input-style__left-content"
        >
           
          <input
            class="m-decimalfield__input"
            id="mDecimalfield-uuid"
            inputmode="numeric"
          />
            
          <!---->
           
          <!---->
        </div>
         
        <div
          class="m-input-style__right-content"
        />
         
        <!---->
      </div>
    </div>
  </div>
   
  <div
    class="m-decimalfield__validation"
  >
    <div
      class="m-validation-message m-decimalfield__validation__message"
    >
      <!---->
       
      <!---->
    </div>
  </div>
</div>
`;

exports[`Storyshots components|m-decimalfield Default precision 1`] = `
<div
  class="m-decimalfield"
  style="width: 100%; max-width: 288px;"
>
  <div
    class="m-input-style m--has-value m--is-tag-default"
  >
    <div
      class="m-input-style__content"
      style="margin-top: 8px;"
    >
      <!---->
       
      <div
        class="m-input-style__body"
      >
        <div
          class="m-input-style__prefix"
        />
         
        <div
          class="m-input-style__left-content"
        >
           
          <input
            class="m-decimalfield__input"
            id="mDecimalfield-uuid"
            inputmode="numeric"
          />
            
          <!---->
           
          <!---->
        </div>
         
        <div
          class="m-input-style__right-content"
        />
         
        <!---->
      </div>
    </div>
  </div>
   
  <div
    class="m-decimalfield__validation"
  >
    <div
      class="m-validation-message m-decimalfield__validation__message"
    >
      <!---->
       
      <!---->
    </div>
  </div>
</div>
`;

exports[`Storyshots components|m-decimalfield Localization 1`] = `
<div
  class="m-decimalfield"
  style="width: 100%; max-width: 288px;"
>
  <div
    class="m-input-style m--has-value m--is-tag-default"
  >
    <div
      class="m-input-style__content"
      style="margin-top: 8px;"
    >
      <!---->
       
      <div
        class="m-input-style__body"
      >
        <div
          class="m-input-style__prefix"
        />
         
        <div
          class="m-input-style__left-content"
        >
           
          <input
            class="m-decimalfield__input"
            id="mDecimalfield-uuid"
            inputmode="numeric"
          />
            
          <!---->
           
          <!---->
        </div>
         
        <div
          class="m-input-style__right-content"
        />
         
        <!---->
      </div>
    </div>
  </div>
   
  <div
    class="m-decimalfield__validation"
  >
    <div
      class="m-validation-message m-decimalfield__validation__message"
    >
      <!---->
       
      <!---->
    </div>
  </div>
</div>
`;

exports[`Storyshots components|m-decimalfield States 1`] = `
<div>
  <div>
    <div
      class="m-decimalfield m--is-disabled"
      style="width: 100%; max-width: 288px;"
    >
      <div
        class="m-input-style m--is-disabled m--is-tag-default"
      >
        <div
          class="m-input-style__content"
          style="margin-top: 8px;"
        >
          <!---->
           
          <div
            class="m-input-style__body"
          >
            <div
              class="m-input-style__prefix"
            />
             
            <div
              class="m-input-style__left-content"
            >
               
              <input
                class="m-decimalfield__input"
                disabled="disabled"
                id="mDecimalfield-uuid"
                inputmode="numeric"
                placeholder="Disabled"
              />
                
              <!---->
               
              <!---->
            </div>
             
            <div
              class="m-input-style__right-content"
            />
             
            <!---->
          </div>
        </div>
      </div>
       
      <div
        class="m-decimalfield__validation"
      >
        <div
          class="m-validation-message m-decimalfield__validation__message"
        >
          <!---->
           
          <!---->
        </div>
      </div>
    </div>
  </div>
   
  <div>
    <div
      class="m-decimalfield m--is-readonly"
      style="width: 100%; max-width: 288px;"
    >
      <div
        class="m-input-style m--is-readonly m--is-tag-default"
      >
        <div
          class="m-input-style__content"
          style="margin-top: 8px;"
        >
          <!---->
           
          <div
            class="m-input-style__body"
          >
            <div
              class="m-input-style__prefix"
            />
             
            <div
              class="m-input-style__left-content"
            >
               
              <input
                class="m-decimalfield__input"
                id="mDecimalfield-uuid"
                inputmode="numeric"
                placeholder="Read-only"
                readonly="readonly"
              />
                
              <!---->
               
              <!---->
            </div>
             
            <div
              class="m-input-style__right-content"
            />
             
            <!---->
          </div>
        </div>
      </div>
       
      <div
        class="m-decimalfield__validation"
      >
        <div
          class="m-validation-message m-decimalfield__validation__message"
        >
          <!---->
           
          <!---->
        </div>
      </div>
    </div>
  </div>
   
  <div>
    <div
      class="m-decimalfield m--is-waiting"
      style="width: 100%; max-width: 288px;"
    >
      <div
        class="m-input-style m--is-waiting m--is-tag-default"
      >
        <div
          class="m-input-style__content"
          style="margin-top: 8px;"
        >
          <!---->
           
          <div
            class="m-input-style__body"
          >
            <div
              class="m-input-style__prefix"
            />
             
            <div
              class="m-input-style__left-content"
            >
               
              <input
                class="m-decimalfield__input"
                disabled="disabled"
                id="mDecimalfield-uuid"
                inputmode="numeric"
                placeholder="Waiting"
              />
                
              <!---->
               
              <!---->
            </div>
             
            <div
              class="m-input-style__right-content"
            />
             
            <div
              class="m-spinner m-input-style__spinner"
            >
              <!---->
               
              <div />
            </div>
          </div>
        </div>
      </div>
       
      <div
        class="m-decimalfield__validation"
      >
        <div
          class="m-validation-message m-decimalfield__validation__message"
        >
          <!---->
           
          <!---->
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Storyshots components|m-decimalfield With initial value (0) 1`] = `
<div
  class="m-decimalfield"
  style="width: 100%; max-width: 288px;"
>
  <div
    class="m-input-style m--has-value m--is-tag-default"
  >
    <div
      class="m-input-style__content"
      style="margin-top: 8px;"
    >
      <!---->
       
      <div
        class="m-input-style__body"
      >
        <div
          class="m-input-style__prefix"
        />
         
        <div
          class="m-input-style__left-content"
        >
           
          <input
            class="m-decimalfield__input"
            id="mDecimalfield-uuid"
            inputmode="numeric"
          />
            
          <!---->
           
          <!---->
        </div>
         
        <div
          class="m-input-style__right-content"
        />
         
        <!---->
      </div>
    </div>
  </div>
   
  <div
    class="m-decimalfield__validation"
  >
    <div
      class="m-validation-message m-decimalfield__validation__message"
    >
      <!---->
       
      <!---->
    </div>
  </div>
</div>
`;

exports[`Storyshots components|m-decimalfield With initial value 1`] = `
<div
  class="m-decimalfield"
  style="width: 100%; max-width: 288px;"
>
  <div
    class="m-input-style m--has-value m--is-tag-default"
  >
    <div
      class="m-input-style__content"
      style="margin-top: 8px;"
    >
      <!---->
       
      <div
        class="m-input-style__body"
      >
        <div
          class="m-input-style__prefix"
        />
         
        <div
          class="m-input-style__left-content"
        >
           
          <input
            class="m-decimalfield__input"
            id="mDecimalfield-uuid"
            inputmode="numeric"
          />
            
          <!---->
           
          <!---->
        </div>
         
        <div
          class="m-input-style__right-content"
        />
         
        <!---->
      </div>
    </div>
  </div>
   
  <div
    class="m-decimalfield__validation"
  >
    <div
      class="m-validation-message m-decimalfield__validation__message"
    >
      <!---->
       
      <!---->
    </div>
  </div>
</div>
`;
